### PR TITLE
chore: add csv example

### PR DESCRIPTION
**Pull request by fansehep**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-rs/issues/1613

# Rationale for this change
 

# What changes are included in this PR?

add a example for arrow-csv.

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-csv/examples/csv_calculation.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use arrow_array::Int16Array;
+use arrow_csv::ReaderBuilder;
+
+use arrow_schema::{DataType, Field, Schema};
+use std::fs::File;
+use std::sync::Arc;
+
+fn main() {
+    // read csv from file
+    let file = File::open("arrow-csv/test/data/example.csv").unwrap();
```
Comment (fansehep) [inline arrow-csv/examples/csv_calculation.rs:27]: Maybe I thnk don't need create a new file.

File: arrow-csv/examples/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Examples
+- [`csv_calculation.rs`](csv_calculation.rs): simple use the csv API to calcuate.
```
Comment (tustvold) [inline arrow-csv/examples/README.md:21]:
```suggestion
- [`csv_calculation.rs`](csv_calculation.rs): performs a simple calculation using the CSV reader
```

File: arrow-csv/examples/csv_calculation.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                println!(
+                    "max value column(0): {:?}",
+                    s.column(0)
+                        .as_any()
+                        .downcast_ref::<Int16Array>()
+                        .unwrap()
+                        .iter()
+                        .max()
+                        .unwrap()
+                );
```
Comment (tustvold) [inline arrow-csv/examples/csv_calculation.rs:52]:
```suggestion
                let col = s.column(0).as_primitive::<Int16Type>();
                let max = col.iter().max().flatten();
                println!("max value column(0): {max:?}");
```
With the necessary additional imports

**Review by tustvold — COMMENTED** [review]
Just some minor nits

**Pull request merged**